app/controllers/users_controller.rb

```diff
@@ -20,37 +20,50 @@ def show
     @edit_projects = Project.where(repo_url: github_user_projects) - @projects
   end
 
-  # rubocop: disable Metrics/MethodLength
+  # rubocop: disable Metrics/MethodLength,Metrics/AbcSize
   def create
     @user = User.find_by(email: user_params[:email])
     if @user
       redirect_existing
     else
       @user = User.new(user_params)
       @user.provider = 'local'
-      if @user.save
-        send_activation
+      if @user.password_valid?(user_params[:password])
+        if @user.save
+          send_activation
+        else
+          render 'new'
+        end
       else
+        flash.now[:danger] = 'Your password does not meet our
+                              complexity/length requirements.'
         render 'new'
       end
     end
   end
-  # rubocop: enable Metrics/MethodLength
+  # rubocop: enable Metrics/MethodLength,Metrics/AbcSize
 
   def edit
     @user = User.find(params[:id])
     redirect_to @user unless @user.provider == 'local'
   end
 
+  # rubocop: disable Metrics/MethodLength,Metrics/AbcSize
   def update
     @user = User.find(params[:id])
-    if @user.update_attributes(user_params)
+    if !@user.password_valid?(user_params[:password])
+      flash.now[:danger] =
+        'Your new password does not meet our
+         complexity/length requirements.'
+      render 'edit'
```

Comment:
If you create a custom validator you should not need this if statement, as the update should fail on its own.